addons/web/static/src/js/fields/basic_fields.js

```diff
@@ -1581,6 +1582,442 @@ var FieldBinaryImage = AbstractFieldBinary.extend({
+            dataType: 'text',
+            headers: {'Content-Type': 'font/ttf'},
+            success: function (res, statut) {
+                var $svg = $(qweb.render('SignatureSvgText', {
+                    width: canvas.width,
+                    height: canvas.height,
+                    font: res,
+                    text: text,
+                    type: this.signatureType,
+                }));
+                $svg.attr({
+                    "xmlns": "http://www.w3.org/2000/svg",
+                    "xmlns:xlink": "http://www.w3.org/1999/xlink",
+                });
+
+                self._printImage("data:image/svg+xml," + encodeURI($svg[0].outerHTML));
+            }
+        });
+    },
+
+    /**
+     * @private
+     * Render and scale the image in the canvas.
+     * @param {string} imgSrc image (either a generated signature or a uploaded image)
+     */
+    _printImage: function (imgSrc) {
+        var self = this;
+
+        var image = new Image;
+        image.onload = function () {
+            var width = 0, height = 0;
+            var ratio = image.width / image.height;
+
+            self.$signatureField.jSignature('reset');
+            var $canvas = self.$signatureField.find('canvas');
+            var context = $canvas[0].getContext("2d");
+
+            if (image.width / $canvas[0].width > image.height / $canvas[0].height) {
+                width = $canvas[0].width;
+                height = width / ratio;
+            } else {
+                height = $canvas[0].height;
+                width = height * ratio;
+            }
+
+            setTimeout(function () {
+                var ignoredContext = _.pick(context, ['shadowOffsetX', 'shadowOffsetY']);
+                _.extend(context, {shadowOffsetX: 0, shadowOffsetY: 0});
+                context.drawImage(image, 0, 0, image.width, image.height, ($canvas[0].width - width) / 2, ($canvas[0].height - height) / 2, width, height);
+                _.extend(context, ignoredContext);
+            }, 0);
+        };
+        image.src = imgSrc;
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     * Select a signature mode: draw manually, automatic generation or upload a signature image
+     * @param {MouseEvent} e
+     */
+    _onClickChangeSignatureMode: function (e) {
+        this.$modeButtons.removeClass('btn-primary');
+        $(e.target).addClass('btn-primary');
+
+        this.mode = $(e.target).data('mode');
+
+        if (SignatureDialog.fonts.length > 1) {
+            this.$selectStyleButton.toggle(this.mode === 'auto');
+        }
+        this.$clearButton.toggle(this.mode === 'draw');
+        this.$loadButton.toggle(this.mode === 'load');
+
+        if (this.mode === 'load') {
+            this.$loadButton.click();
```

Comment:
I would prefer that you do not programmatically click on buttons, but call the method that the button calls instead,if possible.